### PR TITLE
sql: rename MANAGETENANT to MANAGEVIRTUALCLUSTER

### LINKED ISSUE #110927: sql: rename the privilege MANAGETENANT
**Describe the problem**

The privilege is called "MANAGETENANT" but we want this to be useable even without multi-tenant workloads.

**Expected behavior**

Need to abandon the word "TENANT" and find some better name.

Epic: CRDB-29380

Jira issue: CRDB-31677

### PR DESCRIPTION
**Pull request by knz**
Fixes #110927.
Epic: CRDB-29380

Release note (cluster virtualization): the privilege that controls access to CREATE VIRTUAL CLUSTER and other vcluster management syntax is now called MANAGEVIRTUALCLUSTER.

### CODE REVIEW

File: pkg/sql/privilege/privilege.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ const (
 	VIEWJOB                  Kind = 27
 	MODIFYSQLCLUSTERSETTING  Kind = 28
 	REPLICATION              Kind = 29
-	MANAGETENANT             Kind = 30
+	MANAGEVIRTUALCLUSTER             Kind = 30
```
Comment (stevendanna) [inline pkg/sql/privilege/privilege.go:73]: Not sure if this is just a display issue or an actual spacing issue.

**Review by stevendanna — DISMISSED** [review]

**Comment by rafiss** [comment]
is this setting used by anyone currently? if so, there needs to be a migration to convert the name in the system.privileges table

**Comment by stevendanna** [comment]
Uff. Good catch, I completely forgot that these are stored as strings in that table.

Theoretically this was included in the private preview in 23.1.  By the letter of the law we are allowed to break that but we probably shoudn't.

**Comment by yuzefovich** [comment]
For my own edification, where do we write privileges into the system table? On a somewhat quick search I only found a single UPDATE stmt in an existing migration. Do we use direct KV commands?

**Comment by rafiss** [comment]
it is through SQL, but just not that easy to grep for: https://github.com/cockroachdb/cockroach/blob/a499165ea61b1eb5f1d5fec826143126f9ba3258/pkg/sql/grant_revoke_system.go#L139-L149

**Comment by knz** [comment]
@rafiss would the mechanism here #111653 work for you?

**Review by knz — COMMENTED** [review]
Thanks for the rebase!

> Reviewed 11 of 11 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110932#-:-NgNmpL6B5ApyJq4CE2s:b-a1iesb)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)

**Review by yuzefovich — COMMENTED** [review]
RFAL @stevendanna @rafiss 

> Reviewed 8 of 8 files at r1, 4 of 11 files at r5, 41 of 41 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110932#-:-NhI8N8e4KWnowwFXIL9:bhfufl5)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz, @rafiss, and @stevendanna)

**Review by stevendanna — APPROVED** [review]
Thanks for picking this up.

**Comment by yuzefovich** [comment]
TFTR!

bors r+

**Pull request merged**